client/my-sites/pages/page-card-info/index.jsx

```diff
@@ -13,20 +13,62 @@ import Gridicon from 'gridicons';
  * Internal dependencies
  */
 import { isFrontPage, isPostsPage } from 'state/pages/selectors';
+import PostRelativeTimeStatus from 'my-sites/post-relative-time-status';
+import { canCurrentUser } from 'state/selectors';
+import { getEditorPath } from 'state/ui/editor/selectors';
 
-function PageCardInfo( { translate, moment, page, showTimestamp, isFront, isPosts, siteUrl } ) {
+const getContentLink = ( state, siteId, page ) => {
+	let contentLinkURL = page.URL;
+	let contentLinkTarget = '_blank';
+
+	if ( canCurrentUser( state, siteId, 'edit_pages' ) && page.status !== 'trash' ) {
+		contentLinkURL = getEditorPath( state, siteId, page.ID );
+		contentLinkTarget = null;
+	} else if ( page.status === 'trash' ) {
+		contentLinkURL = null;
+	}
+
+	return { contentLinkURL, contentLinkTarget };
+};
+
+function PageCardInfo( {
+	translate,
+	moment,
+	page,
+	showTimestamp,
+	isFront,
+	isPosts,
+	siteUrl,
+	contentLink,
+} ) {
 	const iconSize = 12;
+	const renderFutureTimestamp = function() {
+		if ( page.status === 'future' ) {
+			return (
+				<PostRelativeTimeStatus
```

Comment:
@jeremeylduvall Looks and works great to me. One minor point: all of the other gridicons here are 12px, so the 18px looked just slightly out of place to me. What do you think about updating `PostRelativeTimeStatus` to allow gridicon size to be specified?